gc/base/ParallelDispatcher.cpp

```diff
@@ -757,4 +760,66 @@ MM_ParallelDispatcher::expandThreadPool(MM_EnvironmentBase *env)
 
 	return result;
 }
+
+bool
+MM_ParallelDispatcher::reinitializeThreadPool(MM_EnvironmentBase *env, uintptr_t newPoolSize)
+{
+	 if (!_extensions->isStandardGC()) {
+		  /* Dispatcher reinitialization is only supported for standard collectors. */
+		 Assert_MM_true(newPoolSize <= _poolMaxCapacity);
+	 } else if (newPoolSize > _poolMaxCapacity) {
+		/* Re-size/allocate the dispatcher tables: _threadTable, _statusTable & _taskTable. */
+		 OMR::GC::Forge *forge = env->getForge();
+
+		 omrthread_t *threadTableTemp = (omrthread_t *)forge->allocate(
+				 newPoolSize * sizeof(omrthread_t),
+				 OMR::GC::AllocationCategory::FIXED,
+				 OMR_GET_CALLSITE());
+		 if(NULL == threadTableTemp) {
+			 goto error_no_memory;
+		 }
+		 memset(threadTableTemp, 0, newPoolSize * sizeof(omrthread_t));
+
+		 uintptr_t *statusTableTemp = (uintptr_t *)forge->allocate(
+				 newPoolSize * sizeof(uintptr_t *),
+				 OMR::GC::AllocationCategory::FIXED,
+				 OMR_GET_CALLSITE());
+		 if (NULL == statusTableTemp) {
+			 goto error_no_memory;
+		 }
+		 memset(statusTableTemp, 0, newPoolSize * sizeof(uintptr_t *));
+
+		 MM_Task **taskTableTemp = (MM_Task **)forge->allocate(
+				 newPoolSize * sizeof(MM_Task *),
+				 OMR::GC::AllocationCategory::FIXED,
+				 OMR_GET_CALLSITE());
+		 if (NULL == taskTableTemp) {
+			 goto error_no_memory;
+		 }
+		 memset(taskTableTemp, 0, newPoolSize * sizeof(MM_Task *));
+
+		 for (uintptr_t index = 0; index < _threadCountMaximum; index++) {
+			 threadTableTemp[index] =_threadTable[index];
+			 statusTableTemp[index] = _statusTable[index];
+			 taskTableTemp[index] = _taskTable[index];
+		 }
+
+		 forge->free(_taskTable);
+		 _taskTable = taskTableTemp;
+
+		 forge->free(_statusTable);
+		_statusTable = statusTableTemp;
+
+		forge->free(_threadTable);
+		_threadTable = threadTableTemp;
+
+		_poolMaxCapacity = newPoolSize;
+	}
```

Comment:
is it correct to return true for the `else` case (`newPoolSize` <= `_poolMaxCapacity`)?

Reply:
Yup, I've updated the documentation for this function to explain:

```bool indicating if the thread pool tables can accommodate the newPoolSize (i.e., successfully reallocated or already have capacity). * ```